### ISSUE TITLE
Choose subjects based on an AST diff

### ISSUE THREAD

**Issue by dgollahon** [mbj/mutant#1337]
The --since behavior is great but it has a few rough edges. One of those is that trivial formatting changes (a far-reaching rubocop style change) can trigger a massive amount of mutations. Additionally, there are some issues where changing adjacent lines triggers unchanged subjects.

**Comment by mbj** [comment]
Yes so far mutant does a constraint syntax diff, but it should do an AST diff.

**Comment by dgollahon** [comment]
@mbj bumping this issue since it has been a recent pain point for us.
